### DISCUSSION TITLE
Understanding the structure of the model outputs

### DISCUSSION THREAD

**Discussion by TomasMiskov** [aquacropos/aquacrop#88] [General]
**EDIT:** Using `get_water_storage()`, `get_water_flux()`, and `get_crop_growth()` I can now see the column names of the output arrays.

I've been poking around all of the source files for a day or two but I still can't find a good explanation of what exactly each column in `model._outputs.water_flux`, `model._outputs.water_storage`, and `model._outputs.crop_growth` stand for.

By default, it looks like there are 12 soil compartments (as seen in `soil.py __init__` function where ` dz=[0.1] * 12`), thus I would expect the `water_storage` to have 12 columns but it seems to have 15, what are the additional 3 entries? And how are all the 15 entries order in terms of soil compartments?

As for `water_flux` I suppose these are all potential inflows and outflows. The length of each array is 16. Do we have a reference for each column?

With `crop_growth` there again seem to be 13 different values, is there a reference for these? Even ource file would suffice.

Finally I found this [reference manual](https://github.com/aquacropos/aquacrop-matlab/blob/main/AquaCropOS/Manual/AquaCropOS_v2_UserManual.pdf) for Matlab implementation where some of my questions are answered but the length of the output arrays differ (e.g. in Python I am getting `water_flux` array of length 16 while in the ref. manual they only describe 13 variables).

**Comment by thomasdkelly** [comment]
Hi @TomasMiskov there is an overview in the getting started notebook. https://aquacropos.github.io/aquacrop/notebooks/AquaCrop_OSPy_Notebook_1/

**Comment by thomasdkelly** [comment]
Now that I read the notebook again, maybe there could be some more detail added

**Reply by TomasMiskov** [reply]
Haha, they are for sure a good starting point, especially for some very basic sims, but for more involved use cases one has to dig into the source code. I expect to be working on my thesis for the upcoming half a year using AquaCrop so somewhere in the process I might be able to put together a little more comprehensive "How to query params/variables from your model" notebook. Let's see! Thanks for getting back to me in any case
